Empty `cluster-autoscaler-status` configmap results in panic

<!--
Please answer these questions before submitting your bug report. Thanks!
-->

**Which component are you using?**:

<!--
Which autoscaling component hosted in this repository (cluster-autoscaler, vertical-pod-autoscaler, addon-resizer, helm charts) is the bug in?
-->

**What version of the component are you using?**:


Component version:
v1.26.2

**What k8s version are you using (`kubectl version`)?**:

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.3", GitCommit:"9e644106593f3f4aa98f8a84b23db5fa378900bd", GitTreeState:"archive", BuildDate:"2023-03-19T19:47:22Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.3", GitCommit:"9e644106593f3f4aa98f8a84b23db5fa378900bd", GitTreeState:"clean", BuildDate:"2023-03-15T13:33:12Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
</pre></details>


**What environment is this in?**:

TalosOS 1.3.7

**What did you expect to happen?**:

working/running daemon

**What happened instead?**:

panic

**How to reproduce it (as minimally and precisely as possible)**:

1. Clear out the `cluster-autoscaler-status` configmap
2. restart the cluster-autoscaler

**Anything else we need to know?**:

```
panic: assignment to entry in nil map

goroutine 72 [running]:
k8s.io/autoscaler/cluster-autoscaler/clusterstate/utils.WriteStatusConfigMap({0x5a5ced8, 0xc000461860}, {0x7ffdc3be3c15, 0x12}, {0xc001bc8d80, 0xd39}, 0xc001ab5920, {0x51162d1, 0x19})
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/clusterstate/utils/status.go:94 +0x42f
k8s.io/autoscaler/cluster-autoscaler/core.(*StaticAutoscaler).RunOnce.func1()
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/core/static_autoscaler.go:364 +0x17e
k8s.io/autoscaler/cluster-autoscaler/core.(*StaticAutoscaler).RunOnce(0xc00138b9a0, {0x4?, 0xc000077710?, 0x87e5e60?})
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/core/static_autoscaler.go:657 +0x2f0b
main.run(0xc0004b7900?, {0x5a2aa38, 0xc000aefa10})
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/main.go:449 +0x2cd
main.main.func2({0x5a20c98?, 0xc000684180?})
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/main.go:536 +0x25
created by k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/client-go/tools/leaderelection/leaderelection.go:211 +0x11b
```

--- a/cluster-autoscaler/clusterstate/utils/status.go
+++ b/cluster-autoscaler/clusterstate/utils/status.go
@@ -91,6 +91,9 @@ func WriteStatusConfigMap(kubeClient kube_client.Interface, namespace string, ms
 	maps := kubeClient.CoreV1().ConfigMaps(namespace)
 	configMap, getStatusError = maps.Get(context.TODO(), statusConfigMapName, metav1.GetOptions{})
 	if getStatusError == nil {
+		if configMap.Data == nil {
+			configMap.Data = make(map[string]string)
+		}
 		configMap.Data["status"] = statusMsg
 		if configMap.ObjectMeta.Annotations == nil {
 			configMap.ObjectMeta.Annotations = make(map[string]string)

--- a/cluster-autoscaler/clusterstate/utils/status_test.go
+++ b/cluster-autoscaler/clusterstate/utils/status_test.go
@@ -95,6 +95,17 @@ func TestWriteStatusConfigMapExisting(t *testing.T) {
 	assert.True(t, ti.getCalled)
 	assert.True(t, ti.updateCalled)
 	assert.False(t, ti.createCalled)
+
+	// to test the case where configmap is empty
+	ti.configMap.Data = nil
+	result, err = WriteStatusConfigMap(ti.client, ti.namespace, "TEST_MSG", nil, "my-cool-configmap")
+	assert.Equal(t, ti.configMap, result)
+	assert.Contains(t, result.Data["status"], "TEST_MSG")
+	assert.Contains(t, result.ObjectMeta.Annotations, ConfigMapLastUpdatedKey)
+	assert.Nil(t, err)
+	assert.True(t, ti.getCalled)
+	assert.True(t, ti.updateCalled)
+	assert.False(t, ti.createCalled)
 }
 
 func TestWriteStatusConfigMapCreate(t *testing.T) {

